src/linkml/research-dataset-schema.yaml

```diff
@@ -0,0 +1,354 @@
+id: https://concepts.datalad.org/research-dataset-schema
+name: research-dataset-schema
+title: Research dataset schema
+description: >-
+  This schema draft models a generic research dataset for
+  the purpose of making it findable and understandable for
+  humans, for example via an online data catalog.
+prefixes:
+  afo: http://purl.allotrope.org/ontologies/result#
+  bibo: https://purl.org/ontology/bibo/
+  dcterms: https://purl.org/dc/terms/
+  dlco: https://concepts.datalad.org/ontology/
+  dpv: https://w3id.org/dpv#
+  linkml: https://w3id.org/linkml/
+  obo: https://purl.obolibrary.org/obo/
+  openminds: https://openminds.ebrains.eu/controlledTerms/
+  schema: http://schema.org/
+imports:
+  - types
+  - linkml:types
+  - datasets
+default_prefix: dlco
+default_range: string
+
+
+# SLOTS
+slots:
+  address:
+    slot_uri: schema:address
+    range: string
+    description: >-
+      The physical address of an entity
+  affiliation:
+    slot_uri: schema:affiliation
+    multivalued: true
+    inlined_as_list: true
+    range: string
```

Comment:
I believe this does not work, and it is exactly the case that I outlined previously. The instruction says that objects will be inlined. But at the same time the range is set to `string`, a basic type. One of the two needs to change, I think.

Reply:
Thanks this seems like it could be the issue. My original expectation was that the range referred to whatever was inside the list (when using `inlined_as_list`), so that might be the mistake. Will report back soon.

Reply:
It actually looks like this is not the problem. Rather, it looks like the specification of a union of ranges via `any_of`, together with possibly faulty data, is what's causing the issue. I have :

```
slots:
  has_part:
    slot_uri: dcterms:hasPart
    multivalued: true
    inlined_as_list: true
    any_of:
      - range: File
      - range: ResearchDataset
    description: >-
      Linked entities that form part of a dataset
      such as files or other (sub)datasets
```

and then data:

```
"has_part": [
        {
            "checksum_md5": "e7e2be6b203a221949f05e02fcefd853",
            "content_url": "https://portal.edirepository.org/nis/dataviewer?packageid=knb-lter-pal.219.3&entityid=002f3893385f710df69eeebe893144ff",
            "path_posix": "raw/adelie.csv",
            "size_in_bytes": 23755
        },
        {
            "checksum_md5": "1549566fb97afa879dc9446edcf2015f",
            "content_url": "https://portal.edirepository.org/nis/dataviewer?packageid=knb-lter-pal.220.3&entityid=e03b43c924f226486f2f0ab6709d2381",
            "path_posix": "raw/gentoo.csv",
            "size_in_bytes": 11263
        },
        {
            "checksum_md5": "e4b0710c69297031d63866ce8b888f25",
            "content_url": "https://portal.edirepository.org/nis/dataviewer?packageid=knb-lter-pal.221.2&entityid=fe853aa8f7a59aa84cdd3197619ef462",
            "path_posix": "raw/chinstrap.csv",
            "size_in_bytes": 18872
        }
    ]
```

When I set this up I was thinking "how will the validator know whether an object in the list is a `File`  or a `ResearchDataset`, and I'm guessing this is the issue. It's maybe expecting a ID as string? Will investigate this further.

Reply:
When I change the schema to:

```
slots:
  has_part:
    slot_uri: dcterms:hasPart
    multivalued: true
    inlined_as_list: true
    range: File
    description: >-
      Linked entities that form part of a dataset
      such as files or other (sub)datasets
```

and keep the same data, validation succeeds.

Reply:
Yeah, but the difference is `File` being a class, not a type, which make inlining valid. So you resolved that particular flaw from the other end.